doc: thread: SSED and device type updates (#11619)
nrfconnect/sdk-nrf: doc/nrf/protocols/thread/device_types.rst

   :depth: 2

Thread devices can be either Full Thread Devices (FTD) or Minimal Thread Devices (MTD).
The devices are divided based on the following criteria:

> * How they exist in a Thread network - whether the device keeps its radio enabled all the time.
> * How they route messages - whether it maintains a routing table (can route messages) or not (does not route messages).

Reviewer: What about this?

```suggestion
* What role they have in a Thread network - whether they are router-capable (Router, Leader) or can be just an end-device
* What is the communication scheme - whether the device keeps its radio enabled all the time or sporadically (allowing device to sleep and save power)
* How they route messages - whether it maintains a routing table (can route messages) or not (does not route messages).
* Which Thread management functionalities they implement - whether they perform for example address resolution or address registration by itself or rely on the parent
```

> FTDs are the devices that keep their radios enabled all the time and maintain a routing table, while MTDs do not have their radio enabled at all times and they do not maintain a routing table, consequently forwarding all their messages to a parent instead.

Reviewer: ```suggestion
FTDs are the devices that keep their radios enabled all the time and maintain a routing table, while MTDs may have their radio enabled at all times and they do not maintain a routing table, consequently forwarding all their messages to a parent instead.
```

MED is one of MTD role which keeps its receiver on all the time - so "may" is more propriate.

Full Thread Devices
*******************

FTDs keep their radio enabled at all times to be ready to receive messages asynchronously.
They typically consume more power than MTDs and are mains-powered, which allows them to have the following functionalities that MTDs do not have, among others:

* Ability to resolve IPv6 addresses
* Handling address registration
* Maintaining routing table
* Maintaining links with neighbors

Another important difference between FTDs and MTDs is that FTDs subscribe to special "all-routers" multicast addresses (``ff02::2`` and ``ff03::2``, see `IPv6 multicast addressing`_).

> Full Thread Devices can be further divided into the following categories, which depend on the topology of the network:

Reviewer: ```suggestion
Full Thread Devices can be further divided into the following roles, which depend on the topology of the network:
```

* Router
* Full End Device (FED) or Router Eligible End Device (REED)

.. list-table:: Full Thread Device network role categories
   :widths: 15 10 10 15
   :header-rows: 1

   * -
     - Router
     - Full End Device (FED) or Router Eligible End Device (REED)
   * - Maintains a routing table
     - Yes
     - Yes
   * - Routes messages
     - Yes
     - No
   * - Can extend the network
     - Yes
     - No

Routers can be regarded as the backbone of a Thread network.
They maintain the routing table and forward messages to other devices.
Routers, unlike End Devices, can also be used to extend the network range and a Router is required for a Minimal End Device to join the network.
The maximum number of active Routers in a single Thread network is 32.

> FEDs are End Devices that maintain a routing table, but they can't route messages.

Reviewer: ```suggestion
FEDs are End Devices with FTDs capabilities to perform certain Thread management functionalities such as address resolution. Effectively reducing it's router computation power"
```

In last sentence.. i want to say that FED relieves its partent from doing things on behalf of the child. This is the main advantage of FED.

     - No
     - No
     - No
   * - Radio
     - Always enabled
     - Periodically enabled
     - Periodically enabled
   * - New messages
     - Parent immediately forwards the messages
     - The SED polls for new messages when it wakes up
     - Parent forwards the messages during designated transmission window or when SSED explicitly polls data
   * - No new messages
     - No data transmission
     - Parent indicates no pending messages
     - No data transmission*

MEDs are the most basic MTDs, and their radio is always enabled.
They don't keep a routing table and have a :ref:`more limited range of functionalities than FEDs <thread_types_ftd>`.

SEDs try to limit their power consumption by sleeping most of the time, waking up periodically to poll for messages from their parent.
After waking up, they send a data request to their parent.
If the parent has any pending messages, it will send them to the SED.

> (This can happen at any time, since Routers have their radio enabled all the time.)

Reviewer: nit: Does it make sence to have this information together with previous sentence? I mean...

"[..] to the SED (this can happen at any time..."

After waking up, they send a data request to their parent.
If the parent has any pending messages, it will send them to the SED (which can happen at any time, since Routers have their radio enabled all the time.)
Otherwise, the parent will send a response indicating no pending messages.

SSEDs operate similarly to Sleepy End Devices, but they are synchronized with their parent.
They wake up at designated transmission windows agreed with their their parent, which eliminates the need for polling for messages.
If the parent has messages for the SSED, it sends them during the designated transmission window.
The SSED allows the transmission to finish if radio activity is detected during the transmission window.

.. note::
    If the length of the message exceeds the length of the transmission window, the first frame is received in the designated transmission window, but the rest is transmitted using regular Data Polls, exactly like SED.

Conversely, if there is no radio activity during the specified duration of transmission window, this indicates that the parent has no messages for the SSED and the SSED returns to sleep.

> The SSED synchronization results in lower power consumption compared to an SED, primarily because the SSED doesn't need to poll for messages, keeping transmission windows short.

Reviewer 2: ```suggestion
The SSED synchronization usually results in lower power consumption compared to an SED, primarily because the SSED doesn't need to poll for messages, keeping transmission windows short.
```

After waking up, they send a data request to their parent.
If the parent has any pending messages, it will send them to the SED (which can happen at any time, since Routers have their radio enabled all the time.)
Otherwise, the parent will send a response indicating no pending messages.

SSEDs operate similarly to Sleepy End Devices, but they are synchronized with their parent.
They wake up at designated transmission windows agreed with their their parent, which eliminates the need for polling for messages.
If the parent has messages for the SSED, it sends them during the designated transmission window.
The SSED allows the transmission to finish if radio activity is detected during the transmission window.

.. note::
    If the length of the message exceeds the length of the transmission window, the first frame is received in the designated transmission window, but the rest is transmitted using regular Data Polls, exactly like SED.

Conversely, if there is no radio activity during the specified duration of transmission window, this indicates that the parent has no messages for the SSED and the SSED returns to sleep.

> The SSED synchronization usually results in lower power consumption compared to an SED, primarily because the SSED doesn't need to poll for messages, keeping transmission windows short.

Reviewer 2: ```suggestion
The SSED synchronization results in lower power consumption compared to an SED in some scenarios, primarily because the SSED doesn't need to poll for messages, keeping transmission windows short.
```
Sorry for going over this again, but we should be clear that SSED is not better than SED in power consumption as a general rule.
